### PR TITLE
feat(broker-core): write variable events

### LINKED ISSUE #1619: Write dedicated payload events
Since the workflow instance payload is stored internally as variables per scope, we don't need to pass it as payload in each record. Instead, we should write explicit records for each variable and scope.

AT:
* create a new record for variables which contains the name, the value and the scope of the variable
* write a new event if a variable is created or updated on a scope

### PR DESCRIPTION
**Pull request by saig0**
* create new record for variable changes
* write a variable event if a variable is created or changed

closes #1619

### CODE REVIEW

File: broker-core/src/main/java/io/zeebe/broker/workflow/state/VariablesState.java [hunk 1]
```diff
@@ -114,12 +119,29 @@ private void setVariableLocal(
...
+    final DirectBuffer currentValue = getVariableLocal(scopeKey, name, nameOffset, nameLength);
+
+    if (currentValue == null) {
+      variablesColumnFamily.put(scopeKeyVariableNameKey, newVariable);
+
+      if (listener != null) {
+        listener.onCreate(variableName.getBuffer(), newValue, scopeKey);
+      }
+
+    } else if (!BufferUtil.equals(currentValue, newValue)) {
```
Comment (Zelldon) [inline broker-core/src/main/java/io/zeebe/broker/workflow/state/VariablesState.java]: If we have here a performance problem I will say: I told you. :smile:

File: broker-core/src/test/java/io/zeebe/broker/workflow/variables/WorkflowInstanceVariableTest.java [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+        .hasValue("1");
+  }
+
+  @Test
+  public void shouldCreateMultipleVariablesFromPayload() {
+    // given
+    testClient.deploy(WORKFLOW);
+
+    // when
+    testClient.createWorkflowInstance(PROCESS_ID, "{'x':1, 'y':2}");
```
Comment (Zelldon) [inline broker-core/src/test/java/io/zeebe/broker/workflow/variables/WorkflowInstanceVariableTest.java]: Can we add a test for updating an existing varaible and creating a new. So one variable already exist and the other is newly created

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]

File: broker-core/src/main/java/io/zeebe/broker/workflow/state/VariablesState.java (hunk 1 above)
Comment (saig0) [reply broker-core/src/main/java/io/zeebe/broker/workflow/state/VariablesState.java]: I agree but I think we should take it as a future improvement.

File: broker-core/src/test/java/io/zeebe/broker/workflow/variables/WorkflowInstanceVariableTest.java (hunk 2 above)
Comment (saig0) [reply broker-core/src/test/java/io/zeebe/broker/workflow/variables/WorkflowInstanceVariableTest.java]: Sure, I'll create a new case for this.

File: broker-core/src/main/java/io/zeebe/broker/workflow/state/VariablesState.java (hunk 1 above)
Comment (Zelldon) [reply broker-core/src/main/java/io/zeebe/broker/workflow/state/VariablesState.java]: Do we have an issue for this? :sweat_smile:

**Review by ChrisKujawa — APPROVED** [review]

**Comment by saig0** [comment]
bors retry

**Pull request merged**